extensions/applicationinsights-clickanalytics-js/src/events/PageAction.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * pageAction.ts
+ * @author Krishna Yalamanchili (kryalama)
+ * @copyright Microsoft 2020
+ */
+
+import { WebEvent } from './WebEvent';
+import * as DataCollector from '../DataCollector';
+import { ITelemetryItem, getPerformance } from "@microsoft/applicationinsights-core-js"
+import { IPageActionOverrideValues, IPageActionTelemetry, IPageActionProperties } from '../Interfaces/Datamodel';
+import { _extractFieldFromObject, _bracketIt, isValueAssigned, extend } from '../common/Utils';
+
+export class PageAction extends WebEvent {
+    
+    /**
+     * API to send pageAction event
+     * @param pageActionEvent - PageAction event
+     * @param properties - PageAction properties(Part C)
+     */
+    public trackPageAction(pageActionEvent: IPageActionTelemetry, properties?: IPageActionProperties): void {
+        // Get part A properties
+        var ext = {};
+        ext['web'] = {};
+        ext['web']['isManual'] = pageActionEvent.isManual;
+        let event: ITelemetryItem = {
+            name: 'Ms.Web.PageAction',
+            baseType: 'ClickData',
+            ext,
+            data: {},
+            baseData: {}
+        };
+        event.baseData['name'] = pageActionEvent.name;
+        event.baseData['uri'] = pageActionEvent.uri;
+        event.baseData['pageType'] = pageActionEvent.pageType;   
+        event.baseData['properties'] = pageActionEvent.properties;
+        event.baseData['actionType'] = pageActionEvent.actionType;
+        event.baseData['behavior'] = pageActionEvent.behavior;
+        event.baseData['clickCoordinates'] = pageActionEvent.clickCoordinates;
+        event.baseData['content'] = pageActionEvent.content;
+        event.baseData['targetUri'] = pageActionEvent.targetUri;
+        
+        for (let property in properties) {
+            if (properties.hasOwnProperty(property)) {
+                if (!event.data[property]) {
+                    event.data[property] = properties[property];
+                }
+            }
+        }
+        this._clickAnalyticsPlugin.core.track(event);
+    }
+
+    /**
+     * API to create and send a populated PageAction event 
+     * @param element - DOM element
+     * @param overrideValues - PageAction overrides
+     * @param customProperties - Custom properties(Part C)
+     * @param isRightClick - Flag for mouse right clicks
+     */
+    public capturePageAction(element: Element, overrideValues?: IPageActionOverrideValues, customProperties?: { [name: string]: string | number | boolean | string[] | number[] | boolean[] | object }, isRightClick?: boolean): void {
```

Comment:
ApplicationInsights code doesn't expose any "capture" kind of methods, exposing this could cause inconsistencies and confusion I suggest to remove this and the Override values interfaces